### PR TITLE
[LISKDOCS-12] Add guide for redstone push

### PR DESCRIPTION
**Pull request by Tschakki**

### CODE REVIEW

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+- `getRedStoneLSKDataFeedLatestAnswer()`
+- `getRedStoneUSDTDataFeedLatestAnswer()`
+
+Inside of the functions, call the [latestRoundData](https://docs.chain.link/data-feeds/api-reference#latestrounddata) on the respective data feeds to receive the latest spot prices for the respective token.
+
+The `latestRoundData()` function returns the following values:
+
+- `roundId`(uint80): The round ID.
+The "round" the answer was created in.
+Every time a price feed is updated, it adds +1 to the round ID.
```
Comment (matjazv) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]: We're using data feed contracts without rounds, so round is always `1`:
Name of the contracts tells us this: `L2PriceFeedWithoutRounds`

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+## Data feeds on Lisk
+The following Aggregators are available on Lisk Mainnet for Redstone Push:
+
+- [ETH/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0x6b7AB4213c77A671Fc7AEe8eB23C9961fDdaB3b2)
+  - address: `0x6b7AB4213c77A671Fc7AEe8eB23C9961fDdaB3b2`
+- [LSK/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96)
+  - address: `0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96`
+- [USDT/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD)
+  - address: `0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD`
```
Comment (matjazv) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md:49]: We've just recently deployed two new data feed contracts if you want to add them:
- [USDC/USD](https://blockscout.lisk.com/address/0xb4e6A7861067674AC398a26DD73A3c524C602184)
- [WBTC/USD](https://blockscout.lisk.com/address/0x13da43eA89fB692bdB6666F053FeE70aC61A53cd)

File: docs/building-on-lisk/using-oracle-data/redstone-pull.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 title: ...with Redstone (Pull)
 slug: /building-on-lisk/using-oracle-data/redstone-pull
-description: A guide on using Redstone Data Feeds to access real-world data such as asset prices, directly from your smart contracts on the Lisk testnet.
```
Comment (matjazv) [inline docs/building-on-lisk/using-oracle-data/redstone-pull.md:4]: I think everywhere should be `Redstone` => `RedStone`.

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+:::
+
+## Import
+
+To use the RedStone data inside your contract, import the [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) from Chainlink like shown in the example contract below.
+
+For every data feed you like to store, create a new constant with type `AggregatorV3Interface`.
+
+In the constructor, set the above defined constants to point to the respective data feeds:
+Use the `AggregatorV3Interface()` function and pass the adress of the respective data feed contract as parameter.
```
Comment (matjazv) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
Use the `AggregatorV3Interface()` function and pass the address of the respective data feed contract as parameter.
```

File: i18n/ind/docusaurus-plugin-content-docs/current/building-on-lisk/using-oracle-data/redstone-push.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  ]
+---
+
+# Accessing real-world data using the Redstone Oracle (Push)
+
+This page will explain how you can access oracle data using [Redstone Push](https://docs.redstone.finance/docs/get-started/models/redstone-push).
+
+RedStone is a data ecosystem that delivers frequently updated, reliable, and diverse data for your dApp and smart contracts deployed on Lisk.
+
+RedStone data feeds are compatible with Chainlinks [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) and include the following components:
```
Comment (matjazv) [inline i18n/ind/docusaurus-plugin-content-docs/current/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
RedStone data feeds are compatible with Chainlink's [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) and include the following components:
```

**Review by matjazv — CHANGES_REQUESTED** [review]

**Review by matjazv — APPROVED** [review]
LGTM!
Nice job! 🚀

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+## Import
+
+To use the RedStone data inside your contract, import the [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) from Chainlink like shown in the example contract below.
+
+For every data feed you like to store, create a new constant with type `AggregatorV3Interface`.
+
+In the constructor, set the above defined constants to point to the respective data feeds:
+Use the `AggregatorV3Interface()` function and pass the address of the respective data feed contract as parameter.
+
+```solidity
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]: I would add a snippet title to both code snippets for legibility.

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+This page will explain how you can access oracle data using [RedStone Push](https://docs.redstone.finance/docs/get-started/models/redstone-push).
+
+RedStone is a data ecosystem that delivers frequently updated, reliable, and diverse data for your dApp and smart contracts deployed on Lisk.
+
+RedStone data feeds are compatible with Chainlink's [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) and include the following components:
+
+- **Aggregator contract**: An aggregator is a contract that receives periodic data updates from the oracle network.
+Aggregators store aggregated data onchain so that consumers can retrieve it and act upon it within the same transaction.
+These contracts are already deployed on the Lisk network and can be directly used by Consumers.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
These contracts have already been deployed on the Lisk network and can be directly used by consumers.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+- [LSK/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96)
+  - address: `0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96`
+- [USDT/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD)
+  - address: `0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD`
+- [USDC/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xb4e6A7861067674AC398a26DD73A3c524C602184)
+  - address: `0xb4e6A7861067674AC398a26DD73A3c524C602184`
+- [WBTC/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0x13da43eA89fB692bdB6666F053FeE70aC61A53cd)
+  - address: `0x13da43eA89fB692bdB6666F053FeE70aC61A53cd`
+
+In this guide, we will develop a Consumer contract that will request the latest spot prices from the ETH, LSK and USDT data feeds.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
In this guide, we will develop a Consumer contract that requests the latest spot prices from the ETH, LSK, and USDT data feeds.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+
+:::note
+RedStone Push is only fully available on Lisk Mainnet, so please make sure to deploy your Consumer contract on Lisk Mainnet as well.
+
+In case you wish to deploy on Lisk Sepolia Testnet, check the [Tellor](./tellor.md) guide, which is available for both networks.
+:::
+
+## Import
+
+To use the RedStone data inside your contract, import the [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) from Chainlink like shown in the example contract below.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
To use the RedStone data inside your contract, import the [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) from Chainlink, as shown in the example contract below.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+
+For every data feed you like to store, create a new constant with type `AggregatorV3Interface`.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
Create a new constant with the type `AggregatorV3Interface` for every data feed you want to store.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+
+In the constructor, set the above defined constants to point to the respective data feeds:
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
In the constructor, set the above-defined constants to point to the respective data feeds.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+Use the `AggregatorV3Interface()` function and pass the address of the respective data feed contract as parameter.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
Use the `AggregatorV3Interface()` function and pass the address of the respective data feed contract as a parameter.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+```
+
+## Reading data
+To read the data of the price feeds, we define the following functions in the contract:
+
+- `getRedStoneETHDataFeedLatestAnswer()`
+- `getRedStoneLSKDataFeedLatestAnswer()`
+- `getRedStoneUSDTDataFeedLatestAnswer()`
+
+Inside of the functions, call the [latestRoundData](https://docs.chain.link/data-feeds/api-reference#latestrounddata) on the respective data feeds to receive the latest spot prices for the respective token.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]:
```suggestion
Inside the functions, call the [latestRoundData](https://docs.chain.link/data-feeds/api-reference#latestrounddata) on the respective data feeds to receive the latest spot prices for the respective token.
```

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES HARDCODED
+ * VALUES FOR CLARITY.
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]: Please take this out and put it in a `caution` for better visibility.

**Review by TalhaMaliktz — CHANGES_REQUESTED** [review]

File: docs/building-on-lisk/using-oracle-data/redstone-push.md [hunk 15]
```diff
@@ -132,17 +132,15 @@ In this example, we will only use `answer` and `updatedAt`.
 The `updatedAt` value should be used to make sure that the `answer` is recent enough for your application to use it.
 You can compare `updatedAt` to the latest block time (`uint256 currentTime = block.timestamp;`) to ensure you are only using the latest oracle data in your application.
 
-```solidity
+:::caution
+THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+DO NOT USE THIS CODE IN PRODUCTION.
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/using-oracle-data/redstone-push.md]: Too pronounced! Just sentence case would be fine :-D

**Review by TalhaMaliktz — APPROVED** [review]
All good!

**Pull request merged**